leapp/logger/__init__.py

```diff
@@ -52,39 +52,65 @@ def _remote_emit(self, log_data):
 
 
 def configure_logger(log_file=None):
-    global _logger
-    if not _logger:
+    """
+    Configure loggers as per the description below.
+
+    logger: root
+      level: DEBUG
+      handler: StreamHandler
+        level: based on the debug/verbosity options
+      handler: LeappAuditHandler
+        level: DEBUG
+      logger: urllib3
+        level: WARN
+      logger: leapp
+        level: NOTSET
+        handler: FileHandler
+          level: DEBUG
+
+    :return: The 'leapp' logger
+    """
+    global _leapp_logger
+    if not _leapp_logger:
+        _leapp_logger = logging.getLogger('leapp')
+
         log_format = '%(asctime)s.%(msecs)-3d %(levelname)-8s PID: %(process)d %(name)s: %(message)s'
         log_date_format = '%Y-%m-%d %H:%M:%S'
-        path = os.getenv('LEAPP_LOGGER_CONFIG', '/etc/leapp/logger.conf')
+        logging.Formatter.converter = time.gmtime
 
+        root_logger = logging.getLogger()
+        root_logger.setLevel(logging.DEBUG)
+
+        path = os.getenv('LEAPP_LOGGER_CONFIG', '/etc/leapp/logger.conf')
         if path and os.path.isfile(path):
             logging.config.fileConfig(path)
         else:  # Fall back logging configuration
-            logging.Formatter.converter = time.gmtime
-            logging.basicConfig(
-                level=logging.ERROR,
-                format=log_format,
-                datefmt=log_date_format,
-                stream=sys.stderr,
-            )
+            stderr_handler = logging.StreamHandler(sys.stderr)
+            stderr_handler.setFormatter(logging.Formatter(fmt=log_format, datefmt=log_date_format))
+            root_logger.addHandler(stderr_handler)  # The log level of this StreamHandler is NOTSET at this moment
+
             logging.getLogger('urllib3').setLevel(logging.WARN)
-            handler = LeappAuditHandler()
-            handler.setFormatter(logging.Formatter(fmt=log_format, datefmt=log_date_format))
-            logging.getLogger('leapp').addHandler(handler)
+
+            audit_handler = LeappAuditHandler()
+            audit_handler.setFormatter(logging.Formatter(fmt=log_format, datefmt=log_date_format))
+            audit_handler.setLevel(logging.DEBUG)
+            root_logger.addHandler(audit_handler)
```

Comment:
@vinzenz, should the audit be attached to the root logger? The file handler is attached to the 'leapp' logger.